packages/kbn-openapi-bundler/tests/bundler/remove_props.test.ts

```diff
@@ -108,6 +108,8 @@ describe('OpenAPI Bundler - remove custom x- props', () => {
     expect(bundledSpec.paths['/api/some_api']!.post).not.toMatchObject({
       'x-codegen-enabled': expect.anything(),
     });
+    // As we have switched to using the js-yaml safeDump function, we are stripping
+    // out invalid types like 'undefined' when the spec is written to file.
```

Comment:
To reviewers: does it make sense to refactor this test, or rather to implement unit tests for the `write_yaml_document.ts` utility? The end result is the same, hence this test passes, but the behavior at the utility level is different.

Reply:
In fact `write_yaml_document.ts` contained a hack to remove `undefined` via `JSON.parse(JSON.stringinfy())`. I removed this hack so the result behavior stays the same.

The output of this utility is committed to the repo so we can immediately spot any inconsistencies. It also means that having a lot of unit tests to verify all aspects isn't strictly necessary.